core/server/master/src/main/java/alluxio/master/file/InodePermissionChecker.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file;
+
+import alluxio.exception.AccessControlException;
+import alluxio.master.file.meta.Inode;
+import alluxio.security.authorization.Mode;
+
+import java.util.List;
+
+/**
+ * Checks and gets permitted actions on Inodes.
+ */
+public interface InodePermissionChecker {
+  /**
+   * Checks whether user has the specified permission on the file or directory at path.
+   *
+   * @param user the user
+   * @param groups the groups that user belongs to
+   * @param path the path
+   * @param inode inode of the path
+   * @param permission the permissions to check
+   * @throws AccessControlException when user does not have the permission
+   */
+  void checkPermission(String user, List<String> groups, String path, Inode<?> inode,
+      Mode.Bits permission) throws AccessControlException;
```

Comment:
Does `Mode.Bits` need to be extensible/generalized too?

Reply:
In a future PR, `Mode.Bits` will be replaced by `Actions` which can internally be a `Set<Action>`, `Action` can be `READ`, `WRITE`, `EXECUTE`, `PIN`, `FREE`, etc.

For now, to keep the changes minimum, I'll keep the usage of Mode.Bits.